client/driver/rkt.go

```diff
@@ -523,8 +588,51 @@ func (d *RktDriver) Start(ctx *ExecContext, task *structs.Task) (*StartResponse,
 		waitCh:         make(chan *dstructs.WaitResult, 1),
 	}
 	go h.run()
-	//TODO Set Network
-	return &StartResponse{Handle: h}, nil
+
+	d.logger.Printf("[DEBUG] driver.rkt: retrieving status for pod %q (UUID %s) for task %q with", img, uuid, d.taskName)
+	deadline := time.Now().Add(rktNetworkDeadline)
+	var driverNetwork *cstructs.DriverNetwork
+	var notFirst bool
+	for driverNetwork == nil && time.Now().Before(deadline) {
+		if notFirst {
+			time.Sleep(400 * time.Millisecond)
+		}
+		notFirst = true
+
+		status, err := rktGetStatus(uuid)
+		if err != nil {
+			continue
+		}
+		for _, net := range status.Networks {
+			if !net.IP.IsGlobalUnicast() {
+				continue
+			}
+
+			// Get the pod manifest so we can figure out which ports are exposed
+			var portmap map[string]int
+			d.logger.Printf("[DEBUG] driver.rkt: grabbing manifest for pod %q (UUID %s) for task %q", img, uuid, d.taskName)
+			manifest, err := rktGetManifest(uuid)
+			if err == nil {
+				portmap, err = rktManifestMakePortMap(manifest, driverConfig.PortMap)
+				if err != nil {
+					d.logger.Printf("[DEBUG] driver.rkt: could not create manifest-based portmap for %q (UUID %s) for task %q: %v", img, uuid, d.taskName, err)
+				}
+			} else {
+				d.logger.Printf("[WARN] driver.rkt: could get not pod manifest for %q (UUID %s) for task %q: %v", img, uuid, d.taskName, err)
+			}
+
+			driverNetwork = &cstructs.DriverNetwork{
+				PortMap: portmap,
+				IP:      status.Networks[0].IP.String(),
+			}
+			break
+		}
+	}
+	if driverNetwork == nil {
```

Comment:
In the Docker driver we return an error if `driverConfig.PortMap` is set but no networks could be found. I think the same behavior would be appropriate here.

https://github.com/hashicorp/nomad/blob/master/client/driver/docker.go#L1065-L1070